subsys/bluetooth/host/gatt.c

```diff
@@ -252,44 +252,68 @@ static void sc_store(struct gatt_sc_cfg *cfg)
 	       cfg->data.end);
 }
 
-static void sc_clear(struct gatt_sc_cfg *cfg)
+static void clear_sc_cfg(struct gatt_sc_cfg *cfg)
 {
-	BT_DBG("peer %s", bt_addr_le_str(&cfg->peer));
+	memset(cfg, 0, sizeof(*cfg));
+}
+
+static int bt_gatt_clear_sc(u8_t id, const bt_addr_le_t *addr)
+{
+
+	struct gatt_sc_cfg *cfg;
+
+	cfg = find_sc_cfg(id, (bt_addr_le_t *)addr);
+	if (!cfg) {
+		return 0;
+	}
 
 	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
-		bool modified = false;
+		char key[BT_SETTINGS_KEY_MAX];
+		int err;
 
-		if (cfg->data.start || cfg->data.end) {
-			modified = true;
+		if (cfg->id) {
+			char id_str[4];
+
+			u8_to_dec(id_str, sizeof(id_str), cfg->id);
+			bt_settings_encode_key(key, sizeof(key), "sc",
+					       &cfg->peer, id_str);
+		} else {
+			bt_settings_encode_key(key, sizeof(key), "sc",
+					       &cfg->peer, NULL);
 		}
 
-		if (modified && bt_addr_le_is_bonded(cfg->id, &cfg->peer)) {
-			char key[BT_SETTINGS_KEY_MAX];
-			int err;
+		err = settings_delete(key);
+		if (err) {
+			BT_ERR("failed to delete SC (err %d)", err);
+		} else {
+			BT_DBG("deleted SC for %s (%s)",
+			       bt_addr_le_str(&cfg->peer),
+			       log_strdup(key));
+		}
+	}
 
-			if (cfg->id) {
-				char id_str[4];
+	clear_sc_cfg(cfg);
 
-				u8_to_dec(id_str, sizeof(id_str), cfg->id);
-				bt_settings_encode_key(key, sizeof(key), "sc",
-						       &cfg->peer, id_str);
-			} else {
-				bt_settings_encode_key(key, sizeof(key), "sc",
-						       &cfg->peer, NULL);
-			}
+	return 0;
+}
 
-			err = settings_delete(key);
-			if (err) {
-				BT_ERR("failed to delete SC (err %d)", err);
-			} else {
-				BT_DBG("deleted SC for %s (%s)",
-				       bt_addr_le_str(&cfg->peer),
-				       log_strdup(key));
-			}
+static void sc_clear(struct bt_conn *conn)
+{
+	if (!bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
```

Comment:
Could you flip the main- and else-branches so that you don't have to do the negation for the condition?